### PR TITLE
App config override round 2

### LINKED ISSUE #99: Graceful upgrading and defaulting of application config
Status quo:

0. `app-config` tends to include some configuration that's default, that isn't saying what's different about this app. That is, we're [repeating ourselves](http://c2.com/cgi/wiki?DontRepeatYourself).
1. The base `app-config.js` changes in `uw-frame` releases and those changes often need to be copied into the `app-config` of the frame-based application upon its upgrading frame versions. That is, if we [think of app-config as an API, strictly speaking, the API is changing too much too often in not-strictly-backwards-compatible ways](http://semver.org/).
2. Some user-visible angularjs-portal changes ought to routinely be reflected in frame-based apps for consistency ( ~~sidebar content changes~~, footer link changes) where the app doesn't have a reason to be different, but currently at the least this requires the app to update `uw-frame` versions, re-build, re-deploy. That is, configuration and defaulting is compile-time rather than run-time.

A better future:

0. `app-config` in the frame-based app includes only things the app has an opinion about, cases where the app is special or cases where it emphatically wants the current default even if the default were to change.
1. Changes to accepted defaults and additions of defaults in `uw-frame` "just work" in frame-based applications without frame-based applications having to change anything on upgrade of frame version.
2. Some forseeable changes to defaults are reflected in frame-based apps immediately upon change to MyUW `/web`, without touching the deployed frame-based app.

Ideas:

 * Interpretation layer between app-config and the actual runtime config, so that app config can be considered in the context of its version and mashed up with runtime MyUW services.
 * change strategies so that less of the page is provided "by the app" and more is provided "by the portal", rather than the chrome around the app content being compiled into the frame-based app.
 * split app-config and move around config.  Maybe there's a defaults file and then there's app-config and app-config is an overlay on the defaults. Maybe separate into more smaller config files so that a frame-based app only overwrites the configurations it wants to change and at least gets unvarnished defaults when it compile time overlays on `uw-frame`.
 * compile all the frame-based applications together with angularjs-portal into a single unified single-page-application.  Every MyUW release is an update and deployment of the entire MyUW front end, that is, including the front ends of all the applications "in MyUW".

### PR DESCRIPTION
**Pull request by timlevett**
This is round 2 of `override.js` based on recommendations in #141. Resolves #99

Basically we change everything in `app-config.js` to `.value` instead of `.constant`. Then we have a function that goes through the `OVERRIDE` constant and updates anything that it needs to. Arrays are appended to.

### CODE REVIEW

**Comment by timlevett** [comment]
Right now I'm marking as a minor version bump. Upgrade path is pretty easy, and it doesn't break anything if you still override `app-config.js`. So minor unless anyone thinks otherwise.

File: uw-frame-components/js/override.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+define(['angular'], function(angular) {
+
+  /*Keep in sync with docs/mardown/configuration.md*/
```
Comment (apetro) [inline uw-frame-components/js/override.js]: typo in path: "markdown" rather than "mardown".

**Comment by apetro** [comment]
:+1:

**Comment by vertein** [comment]
:+1:

**Pull request merged**